httpclient-jetty/src/main/java/io/fabric8/kubernetes/client/jetty/JettyHttpClient.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.jetty;
+
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.http.HttpRequest;
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import io.fabric8.kubernetes.client.http.Interceptor;
+import io.fabric8.kubernetes.client.http.StandardHttpRequest;
+import io.fabric8.kubernetes.client.http.WebSocket;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.client.util.InputStreamRequestContent;
+import org.eclipse.jetty.client.util.StringRequestContent;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static io.fabric8.kubernetes.client.http.StandardMediaTypes.APPLICATION_OCTET_STREAM;
+import static io.fabric8.kubernetes.client.http.StandardMediaTypes.TEXT_PLAIN;
+import static org.eclipse.jetty.util.BufferUtil.toArray;
+
+public class JettyHttpClient implements io.fabric8.kubernetes.client.http.HttpClient {
+
+  private final HttpClient jetty;
+  private final WebSocketClient jettyWs;
+  private final Collection<Interceptor> interceptors;
+  private final JettyHttpClientBuilder builder;
+  private final JettyHttpClientFactory factory;
+
+  public JettyHttpClient(JettyHttpClientBuilder builder, HttpClient httpClient, WebSocketClient webSocketClient,
+      Collection<Interceptor> interceptors, JettyHttpClientFactory jettyHttpClientFactory) {
+    this.builder = builder;
+    this.jetty = httpClient;
+    this.jettyWs = webSocketClient;
+    this.interceptors = interceptors;
+    this.factory = jettyHttpClientFactory;
+  }
+
+  @Override
+  public void close() {
+    try {
+      jetty.stop();
+      jettyWs.stop();
+    } catch (Exception e) {
+      throw KubernetesClientException.launderThrowable(e);
+    }
+  }
+
+  @Override
+  public DerivedClientBuilder newBuilder() {
+    return builder.copy();
```

Comment:
Hmmm, this one I missed, but I'm unsure how we should proceed.
I'll give it a better look.

Reply:
OK, I was under the assumption that this was going to be possible:

```java
var factory = new JdkHttpClientFactory();
var client1 = factory.newBuilder().build();
var client2 = client1.newBuilder().followRedirects().build();
```

Now I've realized that the derived client only provides certain overrides that should be applicable per-request.

I guess this can be fixed easilly.

Reply:
Should be fixed now